### ISSUE TITLE
ConcurrentMap_Leapfrog::Hash decl

### ISSUE THREAD

**Issue by markpapadakis** [preshing/junction#5]
I think perhaps 
```cpp
using Hash = typename KT::Hash;
```
is more suitable that current
```cpp
typedef typename turf::util::BestFit<Key>::Unsigned Hash;
```

**Comment by preshing** [comment]
Hi Mark, I'm staying away from C++11 in this library for the time being. It's a bit stubborn but I'd like to keep the door open to older & non-mainstream compilers, even if I don't currently use any myself.

**Comment by markpapadakis** [comment]
Hi Jeff. I didn't mean to imply that it should be declared with using instead of a typedef. Just that the Key type is used indirectly to declare Hash instead of KT::Hash. 

I remembered after I posted the issue that junction currently only supports integers as keys and you are using a nifty trick to hash/dehash among keys and their hash representation.  So I see why that's the case. 

By the way, I used the cuckoomap<> someone mentioned in comments of your blog and it just wasn't worth it -- very high contention, system spent most of time spinning the internal spin locks. It just doesn't scale. Had much, much better results with a simple unordered map implementation protected by a fast RW lock ( used Facebook's folly SharedMutex ). 
I am going to try your leapfrog map and see how it fares -- I am sure it will do much better.
